Mage.Sets/src/mage/cards/e/EverythingComesToDust.java

```diff
@@ -0,0 +1,70 @@
+package mage.cards.e;
+
+import mage.MageObjectReference;
+import mage.abilities.effects.common.ExileAllEffect;
+import mage.abilities.keyword.ConvokeAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.util.CardUtil;
+
+import java.util.HashSet;
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class EverythingComesToDust extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent("creatures except those that share a creature type with a creature that convoked this spell, all artifacts, and all enchantments");
+
+    static {
+        filter.add(EverythingComesToDustPredicate.instance);
+    }
+    public EverythingComesToDust(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{7}{W}{W}{W}");
+
+        // Convoke
+        this.addAbility(new ConvokeAbility());
+
+        // Exile all creatures except those that share a creature type with a creature that convoked this spell, all artifacts, and all enchantments.
+        this.getSpellAbility().addEffect(new ExileAllEffect(filter));
+    }
+
+    private EverythingComesToDust(final EverythingComesToDust card) {
+        super(card);
+    }
+
+    @Override
+    public EverythingComesToDust copy() {
+        return new EverythingComesToDust(this);
+    }
+}
+enum EverythingComesToDustPredicate implements ObjectSourcePlayerPredicate<Permanent> {
+    instance;
+    @Override
+    public boolean apply(ObjectSourcePlayer<Permanent> input, Game game) {
+        Permanent p = input.getObject();
+        if (CardType.ARTIFACT.getPredicate().apply(p, game) || CardType.ENCHANTMENT.getPredicate().apply(p, game)){
+            return true;
+        }
+        if (!CardType.CREATURE.getPredicate().apply(p, game)){
+            return false;
+        }
+        HashSet<MageObjectReference> set = CardUtil.getSourceCostsTag(game, input.getSource(), ConvokeAbility.convokingCreaturesKey, new HashSet<>(0));
```

Comment:
Declare variable as interface `Set`, not implementation `HashSet`.

Also can use `Collections.emptySet()` rather than new HashSet

Reply:
The default needs to be a `HashSet` to get the correct generics information, otherwise the class type protection in `getSourceCostsTag` will complain.

I could use `Set` here, but since the value **must** be a `HashSet` to keep the type system happy I feel like it's better to make things explicit.

Reply:
Right, default of type HashSet is needed within `getSourceCostTags`, sorry.

Generally it's not appropriate to specify implementations for variables, but it's not a big deal in this case since it's an internal variable with limited scope that is readily apparent.